main.go

```diff
@@ -43,19 +43,31 @@ type CLI struct {
 
 func (cli *CLI) build(args ...string) {
 	build_errors := 0
+	active_builders := len(cli.config.Extensions)
 
 	for _, e := range cli.config.Extensions {
-		b := build.NewBuilder(e.Development.RootDir)
-
+		b := build.NewBuilder(e)
 		log.Printf("Building %s, id: %s", e.Type, e.UUID)
 
-		result := b.Build(ctx)
+		ch := make(chan build.Result)
 
-		if result.Success {
-			log.Printf("Extension %s built successfully!", e.UUID)
-		} else {
-			build_errors += 1
-			log.Printf("Extension %s failed to build: %s", e.UUID, result.Error.Error())
+		go b.Build(ctx, func(result build.Result) {
+			ch <- result
+		})
+
+		for result := range ch {
+			active_builders--
```

Comment:
Did you see my comment regarding [WaitGroups](https://gobyexample.com/waitgroups) earlier and how that could resolve manually keeping track of counters?